{"status": "resolved", "priority": "urgent", "title": "TwillTests are broken (since Django 1.2.5 switch)", "milestone": "0.11.04", "superceder": "", "nosylist": ["grumler, paulproteus, pythonian4000"], "assigned": "grumler", "waitingon": "", "keywords": [], "id": "377", "files": [{"url": "http://openhatch.org/bugs/file214/0001-Use-csrf_token-with-all-forms.patch", "author": "grumler"}], "history": [{"message": "Marking this as resolved since tests are now passing!\n\n<a href=\"http://linode2.openhatch.org:8080/job/Test%20all%20apps%20except%20customs%20and\">http://linode2.openhatch.org:8080/job/Test%20all%20apps%20except%20customs%20and</a>\n%20search/231/testReport/\n\nGood stuff - grumler!\n   \n", "author": "palhmbs"}, {"message": "Hi grumler,\n\nThis patch looks good. However, curiously, there's one automated test that still\nfails after the Django 1.2.5 upgrade.\n\n<a href=\"http://linode2.openhatch.org:8080/job/Test%20all%20apps%20except%20customs%20and%20search/227/testReport/\">http://linode2.openhatch.org:8080/job/Test%20all%20apps%20except%20customs%20and%20search/227/testReport/</a>\nsays more about the last test run.\n\nYou can run just that test case by typing:\n\n ./bin/mysite test profile.AddCitationManually\n\nI just took a look, and it seems the problem was caused by a rather obscure\nsituation: The default Django manager for the PortfolioEntry model was\npreventing form validation from working as expected.\n\nI pushed your work:\n<a href=\"http://gitorious.org/openhatch/oh-mainline/commit/a8d78320bca21b2da66f92dfa111189e854806c0\">http://gitorious.org/openhatch/oh-mainline/commit/a8d78320bca21b2da66f92dfa111189e854806c0</a>\n\nand I added a commit on top:\n<a href=\"http://gitorious.org/openhatch/oh-mainline/commit/d1cb381b0bead2b49a1825c768ac049261f15ef7\">http://gitorious.org/openhatch/oh-mainline/commit/d1cb381b0bead2b49a1825c768ac049261f15ef7</a>\n\nand I believe this fixes the issues. Once Hudson's automatic test run is happy,\nI will mark this resolved and deploy.\n   \n", "author": "paulproteus"}, {"message": "Made the changes to the remaining templates. It all seems to work now.\n   \n", "author": "grumler"}, {"message": "Ok, as mentioned by paulproteus in IRC this seems to be related to Django 1.2 and \nCSRF. I'll try to fix this.\n   \n", "author": "grumler"}, {"message": "I can take a look at this first-thing, as soon as I'm done with patch review.\n   \n", "author": "paulproteus"}, {"message": "Twill complains that it can't find the link to \"Log in\". I wonder what is going on.\n   \n", "author": "paulproteus"}]}